targets/TARGET_Cypress/TARGET_PSOC6/mbed_overrides.c

```diff
@@ -102,18 +85,4 @@ void mbed_sdk_init(void)
     /* Enable global interrupts (disabled in CM4 startup assembly) */
     __enable_irq();
 #endif
-
-#if defined (CY_CFG_PWR_SYS_IDLE_MODE)
-    /* Configure the lowest power state the system is allowed to enter
-    * based on the System Idle Power Mode parameter value in the Device
-    * Configurator. The default value is system deep sleep.
-    */
-#if (CY_CFG_PWR_SYS_IDLE_MODE == CY_CFG_PWR_MODE_ACTIVE)
-    rtos_attach_idle_hook(&active_idle_hook);
```

Comment:
Instead of removing this entirely and adding extra checks to hal_sleep to turn it into a no-op, could this be replaced by a call to sleep_manager_lock_sleep? That would be simpler and I believe it would produce more correct results from the sleep/deepsleep statistics mechanism when that is enabled.

Reply:
Mbed OS only supports locking deep sleep, not sleep. The locking of deep sleep ensures all features work: https://github.com/ARMmbed/mbed-os/blob/e77b1d8a17ead93517d8ece952a059b7e15afac1/platform/include/platform/mbed_power_mgmt.h#L45-L47
I think in most use cases, the need to lock regular sleep is comparatively low.

@evedon @kjbracey-arm In terms of functionality/actual use cases, does it make sense to add a lock for regular sleep?

Reply:
I dont see why anyone would lock sleep, as it is very shallow mode (core system clock is disabled).

Reply:
As @0xc0170 said, locking normal sleep makes little sense in actual use cases, so we may not add that to the API.